main.go

```diff
@@ -0,0 +1,357 @@
+			// Warning: variables like build_name and output_directory must exist in packer ariables file like in example
+			// #nosec G204
+			cmd := exec.Command("packer", "build", "-var", "build_name="+image.ImageDir, "-var", "output_directory="+outputDirectory, packerImagePath)
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			if err := cmd.Run(); err != nil {
+				fmt.Printf("Error running packer build: %v\n", err)
+				os.Exit(1)
+			}
+			fmt.Println("Packer build completed successfully.")
+
+			// Todo: Use --node-image-registry flag to pass path to registry.yaml
+			_, err = os.Stat(registryConfigPath)
+			if err != nil {
+				fmt.Println(err.Error())
+				os.Exit(1)
+			}
+			// Get the current working directory
+			currentDir, err := os.Getwd()
+			if err != nil {
+				fmt.Printf("Error getting current working directory: %v\n", err)
+				os.Exit(1)
+			}
+
+			// Path to the image created by the packer
+			// Warning: name of the image created by packer should have same name as the name of the image folder in node-images
+			ouputImagePath := filepath.Join(currentDir, outputDirectory, image.ImageDir)
+
+			// Push the built image to S3
+			if err := pushToS3(ouputImagePath, image.ImageDir, registryConfigPath); err != nil {
+				fmt.Printf("Error pushing image to S3: %v\n", err)
+				os.Exit(1)
+			}
+
+			// Update URL in config.yaml if it is necessary
+			if err := updateURLNodeImages(configFilePath, registryConfigPath, image.ImageDir); err != nil {
+				fmt.Printf("Error updating URL in config.yaml: %v\n", err)
+				os.Exit(1)
+			}
+			// Copy config.yaml to releaseDir as node-images.yaml
+			dest := filepath.Join(releaseDir, "node-images.yaml")
+			if err := copyFile(configFilePath, dest); err != nil {
+				fmt.Printf("Error copying config.yaml to releaseDir: %v\n", err)
+				os.Exit(1)
+			}
+			fmt.Println("config.yaml copied to releaseDir as node-images.yaml successfully!")
+		}
+	default:
+		fmt.Println("Unknown method:", method)
+		os.Exit(1)
+	}
+}
+
+func pushToS3(filePath, fileName, registryConfigPath string) error {
+	// Load registry configuration from YAML file
+	// #nosec G304
+	registryConfigFile, err := os.Open(registryConfigPath)
+	if err != nil {
+		return fmt.Errorf("error opening registry config file: %w", err)
+	}
+	defer registryConfigFile.Close()
+
+	var registryConfig RegistryConfig
+	decoder := yaml.NewDecoder(registryConfigFile)
+	if err := decoder.Decode(&registryConfig); err != nil {
+		return fmt.Errorf("error decoding registry config file: %w", err)
+	}
+
+	// Remove "http://" or "https://" from the endpoint if present cause Endpoint cannot have fully qualified paths in minioClient.
+	registryConfig.Config.Endpoint = strings.TrimPrefix(registryConfig.Config.Endpoint, "http://")
+	registryConfig.Config.Endpoint = strings.TrimPrefix(registryConfig.Config.Endpoint, "https://")
+
+	// Initialize Minio client
+	minioClient, err := minio.New(registryConfig.Config.Endpoint, &minio.Options{
```

Comment:
maybe it makes sense to define this client at the beginning, but this is optional

Reply:
As far as it is optional I will resolve this thread and let it be as it is for now if you are ok with it. Maybe in the future PRs, it will be moved at the beginning.